mite/logoutput.py

```diff
@@ -173,7 +192,66 @@ def min_resp_time(self):
     @property
     def req_sec_mean(self):
         return self._req_total / (self._scenarios_completed_time - self._init_time)
-
+    
+    # @property
+    # def resp_time_standard_deviation(self):
+    #     if not self._resp_time_store:
+    #         return 0
+    #     variance = []
+    #     for resp_time in self._resp_time_store:
+    #         variance.append(math.sqrt((resp_time-self.mean_resp_time)**2))
+    #     std_deviation = sum(variance) / len(self._resp_time_store)
+    #     print(f"resp time is {self._resp_time_store}")
+    #     return std_deviation
+    
+    @property
+    def resp_time_standard_deviation(self):
+        if not self._resp_time_store:
+            return 0
+        return np.std(self.resp_time_array)
+    
+    @property
+    def req_sec_standard_deviation(self):
+        if self._req_total == 0:
+            return 0
+        variance = []
+        for req_sec in self._req_sec_store:
+            variance.append(math.sqrt((req_sec-self.req_sec_mean)**2))
+        std_deviation = sum(variance) / len(self._req_sec_store)
+        return std_deviation
```

Comment:
should we use the python standard library here?  https://docs.python.org/3/library/statistics.html#statistics.pstdev

numpy is a beefy dependency which sometimes has build complications (e.g. on alpine -- i personally hate alpine but sometimes we need to use it).  open coded implementations of numerical algorithms like this one can sometimes have numerical stability issues.  i'm not knowledgable enough to say whether that affects this code, but i always opt for using a library when available because that means it's Someone Else's Problem™ to tune the algorithm.  thankfully python makes this available in the standard library, so we don't need to add a dependency to make use of it.

Reply:
I used numpy since it is faster but come to think of it, this is supposed to do operations on smaller datasets so I guess statistics library should do the job. I'll change this